src/vscode-bicep/src/test/e2e/hover.test.ts

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import * as vscode from "vscode";
+
+import { expectDefined, expectRange } from "../utils/assert";
+import { executeHoverProviderCommand } from "./commands";
+import { readExampleFile } from "./examples";
+
+describe("hover", (): void => {
+  let document: vscode.TextDocument;
+
+  beforeAll(async () => {
+    const content = readExampleFile("101", "vm-simple-linux");
+    document = await vscode.workspace.openTextDocument({
+      language: "bicep",
+      content,
+    });
+
+    await vscode.window.showTextDocument(document);
+  });
+
+  afterAll(async () => {
+    await vscode.commands.executeCommand("workbench.action.closeActiveEditor");
+  });
+
+  it("should reveal type signature when hovering over a parameter name", async () => {
+    const hovers = await executeHoverProviderCommand(
+      document.uri,
+      new vscode.Position(1, 7)
+    );
+
+    expectHovers(hovers, {
+      startLine: 1,
```

Comment:
>1 [](start = 17, length = 1)

These positions are one-based?

Reply:
No, they are 0 based. 1 because the first line is a comment.

Reply:
This is also a Monaco thing (I believe the VSCode editor is based on Monaco). At the start of the file, VSCode shows:
![image](https://user-images.githubusercontent.com/38542602/96631774-9d33a280-12cb-11eb-8569-ecc6cc92686a.png)